rust/lance-encoding/src/encodings/physical/bitpack.rs

```diff
@@ -80,14 +94,20 @@ where
 
         if val.to_i64().unwrap() < 0i64 {
             min_leading_bits = min_leading_bits.map(|bits| bits.min(val.leading_ones() as u64));
+            add_signed_bit = true;
         } else {
             min_leading_bits = min_leading_bits.map(|bits| bits.min(val.leading_zeros() as u64));
         }
     }
 
     return min_leading_bits
-        // +1 added here for the sign bit
-        .map(|leading_bits| arr.data_type().byte_width() as u64 * 8 - leading_bits + 1);
+        .map(|leading_bits| arr.data_type().byte_width() as u64 * 8 - leading_bits)
+        .map(|bits| bits + if add_signed_bit { 1 } else { 0 })
+        .map(|bits| bits.max(1)) // cannot bitpack into < 1 bit
+        .map(|bits| BitpackParams {
+            num_bits: bits,
+            signed: add_signed_bit,
+        });
```

Comment:
Very minor nit: All this mapping is fine but you could avoid it and do something like...

```
let mut min_leading_bits = arr.data_type().byte_width() as u64 * 8 - min_leading_bits?;
if add_signed_bit {
  // Need extra sign bit
  min_leading_bits += 1;
} 
// cannot bitpack into <1 bit
let num_bits = min_leading_bits.max(1);
Some(BitpackParams {
  num_bits,
  signed: add_signed_bit
})
```

This might be slightly more readable.